### DISCUSSION TITLE
not  able to pull image from private repo

### DISCUSSION THREAD

**Discussion by yuvrajalpha** [containrrr/watchtower#978] [General] [answered]
I have login to device using docker loing username and passwd then download images from docker hub account 

run container using docker-compose up -d command 

so i update my test container and push on docker hub ,but watchtower not able to pull that test container on my device.

Below is the msg log for your reference
watchtower | time="2021-06-04T12:04:59Z" level=warning msg="Could not do a head request for \"index.docker.io/regalbeloitcr/test_watchtower1:latest\", falling back to regular pull." container=/test1 image="index.docker.io/repo/container_name:latest"

watchtower | time="2021-06-04T12:04:59Z" level=warning msg="Reason: registry responded to head request with \"401 Unauthorized\", auth: \"Bearer realm=\\\"https://auth.docker.io/token\\\",service=\\\"registry.docker.io\\\",scope=\\\"repository:repo/container_name:pull\\\",error=\\\"insufficient_scope\\\"\"" container=/test1 image="index.docker.io/repo/container_name:latest"

watchtower | time="2021-06-04T12:05:02Z" level=info msg="Unable to update container \"/test1\": Error response from daemon: pull access denied for repo/container_name, repository does not exist or may require 'docker login': denied: requested access to the resource is denied. Proceeding to next."

suggest any solution

**Comment by piksel** [comment, marked as answer]
Follow the steps laid out in the [Private Registries](https://containrrr.dev/watchtower/private-registries/) part of the documentation.

**Reply by yuvrajalpha** [reply]
Link help me out ....thanks @piksel
